cmake/HPX_SetupLibCDS.cmake

```diff
@@ -44,9 +32,16 @@ if(HPX_WITH_LIBCDS AND NOT TARGET LibCDS::cds)
     fetchcontent_populate(libcds)
     set(LIBCDS_CXX_STANDARD ${HPX_CXX_STANDARD})
     add_subdirectory(${libcds_SOURCE_DIR} ${libcds_BINARY_DIR})
+    list(TRANSFORM LIBCDS_SOURCELIST PREPEND "${libcds_SOURCE_DIR}/")
+    set(LIBCDS_SOURCE_DIR ${libcds_SOURCE_DIR})
+    # leave the FOLDER properties in place set_target_properties(cds PROPERTIES
+    # FOLDER "Core") set_target_properties(cds-s PROPERTIES FOLDER "Core")
 
-    set_target_properties(cds PROPERTIES FOLDER "Core")
-    set_target_properties(cds-s PROPERTIES FOLDER "Core")
+    # create an imported target that links to the real libcds so that when we
+    # link to the imported target, we don't get export X depends on cds that is
+    # not in the export set
```

Comment:
Is this warning just about export sets? Does libcds get installed alongside HPX?

Having libcds as a "normal" purely external dependency would remove the burden from us to do it correctly (we might need a `FindLibcds.cmake` module though).

Reply:
This question goes to @biddisco

Reply:
I was hoping nobody would ask about the install rules. libcds is not in the export set of hpx, so it won't get installed alongside it. If we are able to use libcds master, then I'm happy to simply have an external libcds that the user must point the build at using `cmake -DLIBCDS_ROOT=...` which removes the need to install ourselves - but if we require a dedicated branch (currently, we need Alex's `hp_generic` branch, then the in-source build is preferred. For now I'm happy to leave things as they are and ignore the install problem since only us developers are using it. If I can't solve it - then I'll switch to an external libcds install when latest stuff is merged to libcds master.

Reply:
I'm glad I asked then! Yeah, I'm also ok with keeping this as they are at the moment, but I think long term having it as a normal external dependency is the only right thing to do.